src/nix/flake.cc

```diff
@@ -1243,25 +1243,45 @@ struct CmdFlakeShow : FlakeCommand, MixJSON
                 auto showDerivation = [&]()
                 {
                     auto name = visitor.getAttr(state->sName)->getString();
+                    std::optional<std::string> description;
+                    if (auto aMeta = visitor.maybeGetAttr(state->sMeta)) {
+                        if (auto aDescription = aMeta->maybeGetAttr(state->sDescription))
+                            description = aDescription->getString();
+                    }
+                    
                     if (json) {
-                        std::optional<std::string> description;
-                        if (auto aMeta = visitor.maybeGetAttr(state->sMeta)) {
-                            if (auto aDescription = aMeta->maybeGetAttr(state->sDescription))
-                                description = aDescription->getString();
-                        }
                         j.emplace("type", "derivation");
                         j.emplace("name", name);
                         if (description)
                             j.emplace("description", *description);
                     } else {
-                        logger->cout("%s: %s '%s'",
-                            headerPrefix,
+                        auto type =
                             attrPath.size() == 2 && attrPathS[0] == "devShell" ? "development environment" :
                             attrPath.size() >= 2 && attrPathS[0] == "devShells" ? "development environment" :
                             attrPath.size() == 3 && attrPathS[0] == "checks" ? "derivation" :
                             attrPath.size() >= 1 && attrPathS[0] == "hydraJobs" ? "derivation" :
-                            "package",
-                            name);
+                            "package";
+                        if (description && !description->empty()) {
+                            // Trim the string and only display the first line of the description.
```

Comment:
This can be done using `filterANSIEscapes()`:
```
/**
 * Truncate a string to 'width' printable characters. If 'filterAll'
 * is true, all ANSI escape sequences are filtered out. Otherwise,
 * some escape sequences (such as colour setting) are copied but not
 * included in the character count. Also, tabs are expanded to
 * spaces.
 */
std::string filterANSIEscapes(std::string_view s,
    bool filterAll = false,
    unsigned int width = std::numeric_limits<unsigned int>::max());
```
I.e. something like `filterANSIEscapes(s, false, getWindowSize().second())`.

Reply:
@edolstra At this point `getWindowSize()` returns 0,0.

Reply:
Could be running headless in tests. Does it work when calling it interactive?

Reply:
Ah yes, I see values if I run `nix flake show` on the console.

Reply:
> Pretty nice would be to output relative to current terminal? (Or 80-ish if headless)

I think that's what the `getWindowSize` approach is supposed to do unless I'm misunderstanding something.

Reply:
Exactly, I was thinking that it would call getWindowSize, if 0, use 80-ish, otherwise use what was returned. Be careful about some pathological tiny windows!